### PR TITLE
Replacing typed queries with hooks in Site settings and Staff modules

### PR DESCRIPTION
**Pull request by krzysztofwolski**
I want to merge this change because typed queries are outdated and are blocking us from migrating to the new apollo.

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [x] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

File: src/siteSettings/views/index.tsx [hunk 1]
```diff
@@ -35,167 +32,149 @@ export const SiteSettings: React.FC<SiteSettingsProps> = ({ params }) => {
...
-                          shopSettingsInput: {
-                            customerSetPasswordUrl: data.customerSetPasswordUrl,
-                            defaultMailSenderAddress:
-                              data.defaultMailSenderAddress,
-                            defaultMailSenderName: data.defaultMailSenderName,
-                            description: data.description
-                          }
-                        }
-                      });
+  const handleShopSettingsUpdate = async (data: SiteSettingsPageFormData) => {
```
Comment (orzechdev) [inline src/siteSettings/views/index.tsx:105]: Previously this handler had returned errors and consequently had `SubmitPromise` as return type. `SubmitPromise` would be used then by `useForm` hook to indicate change state. Why now errors are not returned?

Comment (krzysztofwolski) [reply src/siteSettings/views/index.tsx:105]: Errors are passed separately in line 137. You can test changes yourself - errors are displayed as intended.

Comment (dominik-zeglen) [reply src/siteSettings/views/index.tsx:105]: It allows form to reset it's dirty status, disabling `save` button after successful update

File: src/siteSettings/views/index.tsx [hunk 2]
```diff
@@ -35,167 +32,149 @@ export const SiteSettings: React.FC<SiteSettingsProps> = ({ params }) => {
...
+  } = useSiteSettingsQuery({
+    displayLoader: true,
+    variables: {}
+  });
+
+  const [authorizationKeyAdd, authorizationKeyAddOpts] = useAuthorizationKeyAdd(
+    {
+      onCompleted: data => {
+        if (data.authorizationKeyAdd.errors.length === 0) {
+          notify({
```
Comment (mmarkusik) [inline src/siteSettings/views/index.tsx:47]: This chunk could be moved to a function since it happens 3 times here 🚔

```
const handleComplete = (errors) => {
 if (errors?.length === 0) {
     notify({
        status: "success",
        text: intl.formatMessage(commonMessages.savedChanges)
     });
     navigate(siteSettingsUrl());
   }
}

File: src/staff/views/StaffList/StaffList.tsx [hunk 3]
```diff
@@ -145,90 +134,94 @@ export const StaffList: React.FC<StaffListProps> = ({ params }) => {
...
+        notify({
+          status: "success",
+          text: intl.formatMessage(commonMessages.savedChanges)
+        });
+        navigate(staffMemberDetailsUrl(data.staffCreate.user.id));
+      }
+    }
+  });
+
+  const handleStaffMemberAdd = async (formData: AddMemberFormData) => {
```
Comment (mmarkusik) [inline src/staff/views/StaffList/StaffList.tsx:149]: Just a thought, but could look nicer with a spread 

``` 
async ({permissionGroups, email, firstName, lastName }: AddMemberFormData) => {
```

```
  addGroups: permissionGroups,
  email,
  firstName,
  lastName,

Comment (krzysztofwolski) [reply src/staff/views/StaffList/StaffList.tsx:149]: Don't want to fix all the things along the way in this PR, just migrate to hooks. This way it won't take an eternity and won't produce too big PRs.

File: src/siteSettings/views/index.tsx (hunk 1 above)
Comment (krzysztofwolski) [reply src/siteSettings/views/index.tsx:105]: Good to know, changing.

File: src/siteSettings/views/index.tsx (hunk 2 above)
Comment (krzysztofwolski) [reply src/siteSettings/views/index.tsx:47]: Dont want to mess with this pattern in this PR.  Each notification should have a different message by the way, but it's out of the scope.

**Review by mmarkusik — APPROVED** [review]

File: src/siteSettings/components/SiteSettingsPage/SiteSettingsPage.tsx [hunk 4]
```diff
@@ -55,7 +54,7 @@ export interface SiteSettingsPageProps {
   onBack: () => void;
   onKeyAdd: () => void;
   onKeyRemove: (keyType: AuthorizationKeyType) => void;
-  onSubmit: (data: SiteSettingsPageFormData) => SubmitPromise;
+  onSubmit: (data: SiteSettingsPageFormData) => void;
```
Comment (dominik-zeglen) [inline src/siteSettings/components/SiteSettingsPage/SiteSettingsPage.tsx:57]: Unnecessary change, probably bugging the view

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

File: src/siteSettings/components/SiteSettingsPage/SiteSettingsPage.tsx (hunk 4 above)
Comment (orzechdev) [inline src/siteSettings/components/SiteSettingsPage/SiteSettingsPage.tsx:57]: For consistency I think it should be also changed
```suggestion
  onSubmit: (data: SiteSettingsPageFormData) => SubmitPromise;
```

**Review by orzechdev — APPROVED** [review]
